File: codegen/results/v002/gemma4_31b_think/20260414_091539/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';
import * as path from 'path';

/**
 * Interface representing the structure of each record in the CSV.
 */
interface Record {
  Name: string;
  Age: number;
  Email: string;
  Score: number;
}

function main() {
  try {
    // Define the path to the input file as specified in the prompt
    const filePath = path.join('input', 'data.csv');

    // Read the file content synchronously using utf8 encoding
    const content = fs.readFileSync(filePath, 'utf8');

    // Split content into lines and filter out any trailing empty lines
    const lines = content.split(/\r?\n/).filter(line => line.trim() !== '');

    // Ensure there is at least a header and one row of data
    if (lines.length < 1) {
      process.stdout.write(JSON.stringify([]));
      return;
    }

    // Extract the data rows, skipping the first line (header)
    const dataRows = lines.slice(1);

    const results: Record[] = dataRows.map(line => {
      // Split by comma. Note: This assumes simple CSV without quoted delimiters.
      const [name, ageStr, email, scoreStr] = line.split(',');

      return {
        Name: name,
        Age: parseInt(ageStr, 10),
        Email: email,
        Score: parseFloat(scoreStr),
      };
    });

    // Output the final array as a JSON string to stdout
    process.stdout.write(JSON.stringify(results, null, 2));
  } catch (error) {
    // The prompt specifies "Do not output anything else", but in a real 
    // production environment, we would handle errors here.
    // Given the strict constraints, we fail silently or let the process crash.
    process.exit(1);
  }
}

// Execute the program
main();

File: codegen/results/v002/gemma4_31b_think/20260414_091539/python/001_csv_to_json/solution.py
import csv
import json
import sys

def solve():
    file_path = 'input/data.csv'
    data_list = []

    try:
        with open(file_path, mode='r', encoding='utf-8') as csvfile:
            # DictReader uses the first row as keys for each row dictionary
            reader = csv.DictReader(csvfile)
            
            for row in reader:
                # Construct the object with the specific types requested
                record = {
                    "Name": str(row["Name"]),
                    "Age": int(row["Age"]),
                    "Email": str(row["Email"]),
                    "Score": float(row["Score"])
                }
                data_list.append(record)
                
        # Output the resulting list as a valid JSON array to stdout
        print(json.dumps(data_list))
        
    except FileNotFoundError:
        # In a real scenario, you might handle this differently, 
        # but for this task we assume the file exists.
        sys.exit(1)
    except Exception as e:
        sys.exit(1)

if __name__ == "__main__":
    solve()